tests/test_admin.py

```diff
@@ -524,3 +569,38 @@ class RegistrationTestCase(TestCase):
     def test_admin_is_registered(self):
         self.assertIn(PageContent, admin.site._registry)
         self.assertTrue(isinstance(admin.site._registry[PageContent], PageContentAdmin))
+
+
+@override_settings(CMS_PERMISSION=True)
+class CMSToolbarTestCase(CMSTestCase):
+    def test_pages_menu_item_url_has_no_params(self):
+        """
+        Create a page and get the toolbar for that page's preview
+        Then check that the page menu item does not have a query string,
+        so as not to trigger filters
+        """
+        user = self.get_superuser()
+        pagecontent = PageVersionFactory(content__template="")
+        toolbar = get_toolbar(
+            pagecontent.content,
+            user=user,
+            toolbar_class=PageAdminToolbar,
+            preview_mode=True,
+        )
+        toolbar.post_template_populate()
+        menu = toolbar.toolbar.get_menu("admin-menu")
+        pagemenu = menu.get_items()[0]
+        self.assertTrue("/en/admin/cms/pagecontent/?language=en", pagemenu.url)
+
+    def test_cmstoolbar_is_replaced(self):
+        """
+        Create a page and check that the PageToolbar has been replaced by the PageAdminToolbar
+        """
+        user = self.get_superuser()
+        page = create_page(
+            title="Test", template="page.html", language="en", created_by=user
+        )
+        self.request = self.get_page_request(page, user, "/")
```

Comment:
Why do you need to do this? you don't actually use self.request anywhere after this. I'm wondering if the one line below (the assert) isn't all you need to test this.